### PR TITLE
Vasanth/quiz points system

### PR DESCRIPTION
**Pull request by VASANTH04032004**
Title: Update QuizScreen to Use Answer Class for Handling Quiz Answers

Description:

This pull request updates the `QuizScreen` to integrate the newly created `Answer` class. The following changes were made:

- Replaced the `String` type for answers with the `Answer` class.
- Updated the logic to handle `Answer` objects, including the selection and display of answers.
- Added a `toString()` method to the `Answer` class for proper display in the UI.

Changes:

- Modified the `QuizScreen` to handle answers using the `Answer` class.
- Adjusted the progress bar and animation logic to work with the updated answer structure.
- Fixed type errors related to answer handling.

Testing:

- Verified that the quiz screen displays answers correctly.
- Ensured that the progress bar updates as expected.
- Confirmed that selecting answers and navigating between questions works correctly.

### CODE REVIEW

File: lib/screen/name_input_screen.dart [hunk 1]
```diff
@@ -6,17 +6,49 @@ class UserNameScreen extends StatefulWidget {
...
-        context,
-        MaterialPageRoute(
-          builder: (context) => HomeScreen(userName: _nameController.text),
-        ),
-      );
+      setState(() {
+        _isButtonPressed = true;
+      });
+
+      Future.delayed(Duration(milliseconds: 200), () {
```
Comment (AnaghaVB) [inline lib/screen/name_input_screen.dart]: Is this delay necessary?

File: lib/screen/name_input_screen.dart [hunk 2]
```diff
@@ -29,44 +61,95 @@ class _UserNameScreenState extends State<UserNameScreen> {
...
+                      ],
+                    ),
+                  ),
+                  ScaleTransition(
+                    scale: _isButtonPressed ? _buttonScaleAnimation : AlwaysStoppedAnimation(1),
+                    child: ElevatedButton(
+                      onPressed: _submitName,
+                      style: ElevatedButton.styleFrom(
+                        backgroundColor: Color(0xFF098EAB),
+                        minimumSize: Size(double.infinity, 50),
```
Comment (AnaghaVB) [inline lib/screen/name_input_screen.dart]: I don't see a necessity of adding minimum size here; you achieve the same by using Padding

**Comment by AnaghaVB** [comment]
Wherever you are setting the size for the Elevated button, change it and use Padding instead

File: lib/screen/name_input_screen.dart [hunk 3]
```diff
@@ -29,44 +56,101 @@ class _UserNameScreenState extends State<UserNameScreen> {
...
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.all(16.0),
+                    child: ScaleTransition(
+                      scale: _isButtonPressed ? _buttonScaleAnimation : AlwaysStoppedAnimation(1),
+                      child: ElevatedButton(
+                        onPressed: _submitName,
+                        style: ElevatedButton.styleFrom(
+                          backgroundColor: Color(0xFF098EAB),
+                          minimumSize: Size(double.infinity, 60),
```
Comment (AnaghaVB) [inline lib/screen/name_input_screen.dart:130]: Why using minimumSize?

Comment (AnaghaVB) [inline lib/screen/name_input_screen.dart:130]: Why are you using minimum size here?

Comment (VASANTH04032004) [reply lib/screen/name_input_screen.dart:130]: Using `minimumSize` ensures that the `ElevatedButton` expands to a specified height and width regardless of its content, providing consistent sizing across different screen sizes and layouts. This approach guarantees a uniform appearance for the button in the UI.

**Review by AnaghaVB — APPROVED** [review]

**Pull request merged**